src/diffusers/schedulers/scheduling_lms_discrete.py

```diff
@@ -131,22 +131,24 @@ def lms_derivative(tau):
 
         return integrated_coeff
 
-    def set_timesteps(self, num_inference_steps: int):
+    def set_timesteps(self, num_inference_steps: int, device: Union[str, torch.device] = None, **kwargs):
         """
         Sets the timesteps used for the diffusion chain. Supporting function to be run before inference.
 
         Args:
             num_inference_steps (`int`):
                 the number of diffusion steps used when generating samples with a pre-trained model.
+            device (`str` or `torch.device`, optional):
+                the device to which the timesteps should be moved to. If `None`, the timesteps are not moved.
         """
         self.num_inference_steps = num_inference_steps
 
         timesteps = np.linspace(0, self.config.num_train_timesteps - 1, num_inference_steps, dtype=float)[::-1].copy()
         sigmas = np.array(((1 - self.alphas_cumprod) / self.alphas_cumprod) ** 0.5)
         sigmas = np.interp(timesteps, np.arange(0, len(sigmas)), sigmas)
         sigmas = np.concatenate([sigmas, [0.0]]).astype(np.float32)
-        self.sigmas = torch.from_numpy(sigmas)
-        self.timesteps = torch.from_numpy(timesteps)
+        self.sigmas = torch.from_numpy(sigmas).to(device=device)
+        self.timesteps = torch.from_numpy(timesteps).to(device=device)
```

Comment:
Why is this only required for LMS scheduler ?

Reply:
Also note that with LMS we don't pass `timesteps` to the model but rather index of timesteps, cf https://github.com/huggingface/diffusers/blob/84b9df57a7c78e1cd9c132d286341451a4e6a80b/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L268

Reply:
> Why is this only required for LMS scheduler ?

Because it's the only scheduler that has `self.timesteps` as a torch tensor

Reply:
> Also note that with LMS we don't pass timesteps to the model but rather index of timesteps, cf

The issue i was tackling comes from the UNet https://github.com/huggingface/diffusers/blob/84b9df57a7c78e1cd9c132d286341451a4e6a80b/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L259

Reply:
Ahh I see, my bad! Should we maybe add `.to` method to schedulers which will will put all the scheduler state on the `device` rather than manually handling it in multiple places. This will also avoid creating unused `kwargs` args.
wdyt @patrickvonplaten @anton-l @pcuenca

Reply:
@anton-l also suggested that, but since there were other tensor `.to()` operations in the same scheduler we postponed the decision: https://github.com/huggingface/diffusers/pull/534#issuecomment-1258274317

I think most schedulers don't need `.to()`, but if it saves us from a `kwargs` argument then I think that tips the scale.